Update/Expand What Section of README.md issue #9

nelsonic opened the pull request:
This Pull Request Addresses the question raised by @Cleop in issue #9 💭 

+ [x] Updated [**`README.md`**](https://github.com/dwyl/product-roadmap/blob/update-what-section-issue%239/README.md) to add clarification in the "What" section for #9 
+ [x] Added [**`capture.md`**](https://github.com/dwyl/product-roadmap/blob/update-what-section-issue%239/capture.md) to describe the MVP version of the _first_ step/screen in our app for #12 
+ [x] Tidy up (_streamline_) [**`How?`**](https://github.com/dwyl/product-roadmap/blob/update-what-section-issue%239/README.md#how) section of readme for #16 

I feel these docs are more readable now.
@iteles Please review and give feedback if possible.

nelsonic committed:
add brain-dump.md file with inital contents for https://github.com/dwyl/product-roadmap/issues/12

nelsonic committed:
[WiP] start extending "What?" section for #9 > "Capture, Categorise, Complete"

nelsonic committed:
add background detail for "Brain Dump" issue https://github.com/dwyl/product-roadmap/issues/12

nelsonic committed:
attempt to clarify similar features of existing apps for #9

nelsonic committed:
expand clarification of steps in @dwyl workflow for #9

nelsonic committed:
comment out (distracting) list of universal human needs for #9

nelsonic committed:
add link to "Categorise" issue https://github.com/dwyl/product-roadmap/issues/15

nelsonic committed:
rename brain-dump.md to capture.md consistent with the "Workflow" #9

nelsonic committed:
reduce size of brain dump image #12

nelsonic committed:
update "USP" section of README.md for #9

nelsonic committed:
add "How you can help" section to README.md

nelsonic committed:
simplify "How?" section #16

nelsonic committed:
add enough to product roadmap to get started working on the app! #16

nelsonic commented:
@iteles if/when you have time, please review and give any feedback. Thanks. 📥

nelsonic committed:
move "Why other companies dont open source their product map?" question/answer to end of README.md as it breaks the flow #9

iteles commented on README.md:
```diff
@@ -5,26 +5,25 @@
-on working towards the collective ***goal***.
+Having a _clear_ **Product Roadmap** is ***essential***
+to help ***everyone*** _focus_
+on our collective **goal**.
```
:+1: A fan of how succinct this 'why' is

nelsonic replied:
Yeah, had to tighten it. 👍

iteles commented on README.md:
```diff
@@ -5,26 +5,25 @@
 and/or guarding the user-feedback,
-they are _stuck_ in the _old_ mindset.
-
+they are _stuck_ in a _fixed_ mindset.
```
I want to add something in here about how the goal is not to safeguard our own profits or ability to make money. We need to be _sustainable_ but the more people working towards a world where people are _not_ overwhelmed, riddled with anxiety and always lifting each other up, the better.

nelsonic replied:
@iteles agree with the _sentiment_ of putting people before profit while being sustainable. 👍 
I feel that is more suited to https://github.com/dwyl/start-here/blob/master/manifesto.md 💭

iteles commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
 Public on GitHub is a good or bad idea, we think it's
 worth the _experiment_ just to discover the outcome.
 
-### Why Do So _Few_ Companies Make Their "Road Map" Public?
```
I like that this was removed. I think it distracts from the main message and the two paragraphs above say all that is needed in a much more concise way 👍

nelsonic replied:
@iteles _moved_ to the end of the doc; not _removed_ entirely.

iteles commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+  whereas a reading list item could be "daily commute"<br />
+  **d)** **Add** any additional ***detail*** you can think of at the time.
+  e.g: acceptance criteria, deadline, dependencies <br />
+  **d)** If the item is a task that has sub-tasks, create and link them.
```
This should be `e`.
Also, Linking them definitely belongs here but splitting a task out into actionable steps doesn't _really_ belong here. It will evolve.

nelsonic replied:
@iteles indeed it will evolve _rapidly_. 🐎 
My aim was merely to address #9 and _clarify_ _What_ we are building in the ***MVP*** phase. 👍

iteles commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+  **d)** **Add** any additional ***detail*** you can think of at the time.
+  e.g: acceptance criteria, deadline, dependencies <br />
+  **d)** If the item is a task that has sub-tasks, create and link them.
+3. ***Complete*** - do the work to _complete_ the item/task.<br />
```
This is a very big section. Whereas the first 2 'C's are small, independent actions, this one can involve a lot of different steps and take a long time depending on dependencies (i.e. if you're not doing the work yourself or waiting for bureaucracies). We'll eventually need to split this out but it doesn't have to be done now.

nelsonic replied:
@iteles yeah, the first two steps are relatively succinct and the third is where the _real_ work is done.
I _very_ much want the [**Product Manager/Owner**](https://github.com/dwyl/product-owner-guide/issues/46) to take responsibility for the "Workflow" and get feedback from people _using_ the app. Ultimately the App will succeed (_or fail_) based on the _simplicity_ (_or complexity_) of the workflow (_getting the text of a "brain dump" to be a task/reminder/etc. and then getting it to **`done`** as effectively as possible_)
![image](https://user-images.githubusercontent.com/194400/68548982-fa03b900-03ea-11ea-9cd2-7d763d88b31f.png)

iteles commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+  and determine if you have everything you need to start the task.
+  If not, write down what is missing. <br />
+  **b)** ***Delegate*** if necessary/possible <br />
+  **c)** ***Track progress*** of the task. <br />
```
Estimates and Timers included in this section?

nelsonic replied:
@iteles I'm _so glad_ you asked ... 🎉 
Time estimation and tracking will _absolutely_ be an integral step of ***Categorise*** and ***Complete*** steps in the _second_ sprint of our MVP. 😉 
Want track down (_or create_) the user story to ensure it's in the backlog? 💭

iteles commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+e.g:
++ Linking to external sources of information.
++ Photo/image uploads
+  + Optical Character Recognition (OCR)
```
🙌

nelsonic replied:
Indeed ... https://github.com/tesseract-ocr/tesseract 👀 
we will need to _hire_ someone to do this work after our MVP is working to the PM/PO's and beta tester's satisfaction. i.e. after several other features are complete. But it's definitely _on_ the Roadmap.

iteles commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+that is _universally accessible_ to every human being
+that allows _everyone_ a fair opportunity to live their best life.
+We will _cooperate_ with _many_ people/organisations
+to build a comprehensive solution that _anyone_ can use.
```
This will require clarification. We don't want people to look at this and think 'it's an app that will try to be all things to all people and therefore is guaranteed to fail'. We need to add a section in here about a targeted approach and first audience. I'm just adding in these comments as I'm reading through to make sure I can come back to this PR and make any changes later 😊

nelsonic replied:
Indeed. I'd be _very_ happy to see your (_and other people's_) improvements ... 
https://github.com/dwyl/product-roadmap/graphs/contributors 
![ronery](http://giphygifs.s3.amazonaws.com/media/69ZF2NGvIi0Uw/giphy.gif)

iteles commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+will be _personalised_ to the individual
+based on the features they (_most_) _use_.
+If the person does not _use_ a particular function, they won't _see_ it.
+> Note: This is might not feel relevant during "MVP",
```
🙌 This will of course be coupled with a discovery section where people can take a look at other features

nelsonic replied:
Agreed. That's going to be one of our USPs. Each _feature_ will have a walkthrough demo/video that will explain _why_ the feature exists, _who_ it's for and _how_ to use it. 💡

iteles commented on README.md:
```diff
@@ -132,10 +264,69 @@ which **features** get built **next**_!
+1. **Use** the **_Beta_ App**!
+`[insert app link here!]`
+2. Give us ***feedback*** on the **_existing_ UX**
+`[insert feedback link here!]`
```
🎉

nelsonic replied:
Coming _very_ soon ... 😉

iteles commented on README.md:
```diff
@@ -132,10 +264,69 @@ which **features** get built **next**_!
 
+#### MVP Basic Workflow
+
++ [ ] Mobile First UI/UX to **Capture** **`plaintex`**
```
`plaintext` (also needs correcting in the lines below)

nelsonic replied:
`plaintex` -> **`plaintext`** typos fixed in 3 instances. thanks for spotting. 👍

iteles commented on capture.md:
```diff
@@ -0,0 +1,111 @@
+of everything that is occupying space and processing power
+is _essential_ for gaining focus
+on the _one_ most important priority.
+This step is often referred to as a "brain dump".
```
I would add a note here also about how having a trusted place to keep all of your thoughts is the only way to truly empty your mind. If you have scattered post-it's or a system that makes it difficult to view what needs to be done, the fear of forgetting something keeps everything in RAM anyway.

nelsonic replied:
@iteles yes, please refine as you see fit. 📝 
My objective was just to get _something_ down for the PM/PO to work with. 👍

iteles commented on capture.md:
```diff
@@ -0,0 +1,111 @@
+of capturing free text (_using a **`textarea`** element_)
+and safely storing it as **`text`** in Postgres.
+
+Figma wireframe: https://www.figma.com/proto/WrpkKGJNYZbeCzMRDVtvQLbC/dwyl-app
```
How do we signal that we have _finished_ typing and want to move onto the next item?

nelsonic replied:
@iteles _great_ question! Please capture it in an issue: https://github.com/dwyl/app/issues

iteles reviewed (approved):
@nelsonic This is fantastic, thank you for **capturing** all this (hahaha)

There are 2 comments about typos but the rest of my comments are just notes about things I would like to add myself in the coming days. 

Do you want to correct these 2 typos _before_ I merge the PR? Or shall I go ahead and merge and then I can correct them asap when submitting my next PR?

nelsonic committed:
fix enumeration d) -> e) issue noted by @iteles review https://github.com/dwyl/product-roadmap/pull/13#discussion_r344503002

nelsonic committed:
fix typo plaintex > plaintext noted by @iteles in https://github.com/dwyl/product-roadmap/pull/13#discussion_r344506640

nelsonic commented:
@iteles thanks for reviewing my (_fairly large_) PR in such a quick turnaround. 🎉 
I have fixed the typos you noted and addressed all your comments. ✅ 
Please feel free to _merge_ if you feel it's ready 👍 
***or*** assign to @SimonLab for his thoughts ahead of "Sprint Zero". 😮 

![thank-you](https://media.giphy.com/media/WnIu6vAWt5ul3EVcUE/giphy.gif)

iteles commented:
@SimonLab Let us know your initial thoughts (no need to spend too much time on this, it will change quickly!) ☺️

SimonLab commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+For the first step (***Capture***),
+we will build an ultra-simple interface
+to allow people to
+input **`palintext`**
```
```suggestion
input **`plaintext`**
```

SimonLab commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+We will add additional methods
+to help people "capture" information.
+e.g:
++ Linking to external sources of information.
```
```suggestion
+ Linking to external sources of information
```
Removing the dot to be consistent with the other points. Or should we add them to the other lines?

SimonLab commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+to apply textual labels to items.
+The UX for this is still "TBD".
+We will need to sketch it out ahead of sprint planning.
+See: https://github.com/dwyl/product-roadmap/issues/15
```
```suggestion
See: https://github.com/dwyl/app/issues/235
```

This issue has now been moved to the dwyl/app repo (github is doing the redirection automatically but it might still be worth to have the correct link)

nelsonic committed:
fix typo palintext > plaintext https://youtu.be/psitOvKq6X8

Co-Authored-By: Simon <labdevsimon@gmail.com>

SimonLab commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+
+And finally for the _third_ step, ***Complete***,
+we will build a simple checklist system.
+See: [`checklists.md`](https://github.com/dwyl/product-roadmap/blob/master/checklists.md)
```
I've noticed a few typos on the checklist document, I'll create a new PR to fix them.

nelsonic replied:
@SimonLab yes, please do fix when you have time. thanks. 👍

nelsonic committed:
Inês prefers bullet points to *not* have a full stop, so remove.

Co-Authored-By: Simon <labdevsimon@gmail.com>

nelsonic committed:
update link to "categorise" issue in light of moving to App repo.

Co-Authored-By: Simon <labdevsimon@gmail.com>

SimonLab commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+
+
+
+### USP: What is _different_ about the @dwyl App?
```
![image](https://user-images.githubusercontent.com/6057298/68585420-1a2e8900-047a-11ea-8caa-cdab461e180e.png)

Sub title not displayed correctly. Looking why it's broken.

Also the `USP` meaning wasn't obvious to me and had too google it. We can write it down fully (Unique Selling Point/Proposition) or link it to the wikipedia page: https://en.wikipedia.org/wiki/Unique_selling_proposition

nelsonic replied:
@SimonLab please edit it to help clarify it for others. thanks. 👍

SimonLab replied:
```suggestion
### [USP](https://en.wikipedia.org/wiki/Unique_selling_proposition): What is _different_ about the @dwyl App?
```

SimonLab replied:
Sorry just saw the link to the Wikipedia page just under.

SimonLab commented on README.md:
```diff
@@ -33,65 +32,198 @@ Only time will tell if making our product roadmap
+will be _personalised_ to the individual
+based on the features they (_most_) _use_.
+If the person does not _use_ a particular function, they won't _see_ it.
+> Note: This is might not feel relevant during "MVP",
```
```suggestion
> Note: This might not feel relevant during "MVP",
```
`This is might not` sounds weird to me, no?

SimonLab commented on README.md:
```diff
@@ -132,10 +264,69 @@ which **features** get built **next**_!
+2. Give us ***feedback*** on the **_existing_ UX**
+`[insert feedback link here!]`
+3. **Read** the
+[issues](https://github.com/dwyl/time/issues?q=label%3A%22help+wanted%22+is%3Aissue+is%3Aopen)
```
```suggestion
[issues](https://github.com/dwyl/app/issues?q=label%3A%22help+wanted%22+is%3Aissue+is%3Aopen)
```

Issue moved from time repo to app repo

SimonLab commented on README.md:
```diff
@@ -132,10 +264,69 @@ which **features** get built **next**_!
++ [ ] Save **`plaintext`** data to Phoenix backend
++ [ ] Simple UX to **Categorise** (Transform)
+the **`plaintext`** into actionable items. <br />
+UI/UX **`help wanted`**: https://github.com/dwyl/product-roadmap/issues/15
```
```suggestion
UI/UX **`help wanted`**: https://github.com/dwyl/app/issues/235
```

SimonLab commented on README.md:
```diff
@@ -132,10 +264,69 @@ which **features** get built **next**_!
+#### Offline Support
+
++ [ ] Create Progressive Web App using Elm.
+see:
```
```suggestion
```

Or did you have a link in mind for PWA with Elm?

nelsonic replied:
@SimonLab good catch. The link is: https://github.com/dwyl/learn-elm/issues/54

SimonLab replied:
```suggestion
see: dwyl/learn-elm#54
```

nelsonic replied:
@simonlab thanks for adding the reference. it would be more effective if it were a hyperlink.
`[dwyl/learn-elm#54](https://github.com/dwyl/learn-elm/issues/54)`

SimonLab replied:
Yes indeed. I thought Github was doing it automatically :thinking: 
```suggestion
see: [dwyl/learn-elm#54](https://github.com/dwyl/learn-elm/issues/54)
```

SimonLab commented:
As you've seen :arrow_up: I've added a few suggestions to fix some typos.

I'm not sure yet about the `USP` title yet but hopefully it will be displayed properly.
The other thing I've noticed is the link to the "work hard every waking hour" youtube video is not working anymore:
![image](https://user-images.githubusercontent.com/6057298/68587310-b6f32580-047e-11ea-8a2e-01e22e031a12.png)

We can either remove the link or try to find a new link to the video.

Otherwise looks good :+1:

SimonLab commented on capture.md:
```diff
@@ -0,0 +1,111 @@
+# _Capture_
+
+The **_first_ step** in creating
+a **system** of for personal effectiveness
```
```suggestion
a **system** for personal effectiveness
```

SimonLab commented on capture.md:
```diff
@@ -0,0 +1,111 @@
+First we need to have an ultra-basic MVP working.
+
+
+Please see: https://github.com/dwyl/product-roadmap/issues/12
```
```suggestion
Please see: https://github.com/dwyl/app/issues/234
```

nelsonic committed:
fix grammar in sentence: "This is might" > "This might". Thanks @simonlab :+1:

Co-Authored-By: Simon <labdevsimon@gmail.com>

iteles committed:
Update link to issue in readme

Co-Authored-By: Simon <labdevsimon@gmail.com>

iteles committed:
Update link to "categorise" issue in readme

Co-Authored-By: Simon <labdevsimon@gmail.com>

iteles committed:
Update capture.md grammar 'of' > 'for'

Co-Authored-By: Simon <labdevsimon@gmail.com>

iteles committed:
Updates link to "capture" first issue in capture.md

Co-Authored-By: Simon <labdevsimon@gmail.com>

nelsonic committed:
add link to Elm PWA thread https://github.com/dwyl/learn-elm/issues/54

Co-Authored-By: Simon <labdevsimon@gmail.com>

nelsonic commented:
@SimonLab thanks for taking the time to review this PR in such great detail and propose changes. 👍 
Lame that the video link is broken ...  😞 
Alternative link to Elon Musk "work hard every waking hour": https://youtu.be/liJbB_0eCTo?t=25 📺 

> I'm a fan of Elon's work ethic and feel that it will inspire/motivate others in our community,
so I'm keen to keep the reference. I'm _really_ not a fan of people who do the _bare minimum_ work and think it's important to set expectations. **Note**: Not saying that _everyone_ needs to work 80-100 hours per week like Elon. People should decide for themselves what they want in life. If they want "balance" in their 20's _more_ than _early retirement_ in their 30's that's their decision. Nothing "wrong" with it. 👍

SimonLab reviewed (approved):
Looks great, @iteles I'll let you check one last time if you want and merge the PR?

nelsonic committed:
update link to Elon Musk "work hard every waking hour" https://youtu.be/liJbB_0eCTo?t=26

SimonLab committed:
fixing subtitle of UPS

SimonLab committed:
merge video link

iteles commented:
:+1: Thanks both!
